### PR TITLE
Work on queue of videos

### PR DESCRIPTION
**Pull request by bakerac4**

### CODE REVIEW

File: pep/app/pods/admin/general/controller.ts [hunk 1]
```diff
@@ -49,6 +50,64 @@ export default class AdminGeneral extends Controller {
...
+     * Create the topical lvideo object and add it to the changeset
+     *
+     * @param {VideoConfiguration} video
+     * @memberof AdminGeneral
+     */
+    @action
+    createTopicalVideo(video: VideoConfiguration): void {
+        const previews = this.changeset?.get('configSettings.global.cards.topicalVideoPreviews');
+        previews.push(video);
+        this.changeset?.set('configSettings.global.cards.topicalVideoPreviews', [...previews]);
```
Comment (billdami) [inline pep/app/pods/admin/general/controller.ts:95]:
```suggestion
        this.changeset?.set('configSettings.global.cards.topicalVideoPreviews', [...previews, video]);
```
I think this is a little cleaner for imutable array pushing, nothing to hold up the PR tho.

File: pep/app/pods/admin/general/controller.ts [hunk 2]
```diff
@@ -75,12 +134,38 @@ export default class AdminGeneral extends Controller {
...
+     * Delete the video
+     *
+     * @param {VideoConfiguration} video
+     * @memberof AdminGeneral
+     */
+    @action
+    deleteVideo(video: VideoConfiguration): void {
+        const videos: VideoConfiguration[] = this.changeset?.get('configSettings.global.cards.videoPreviews');
+        const newVideos = videos.filter((item) => item !== video);
+        this.changeset?.set('configSettings.global.cards.videoPreviews', [...newVideos]);
```
Comment (billdami) [inline pep/app/pods/admin/general/controller.ts:147]:
```suggestion
        this.changeset?.set('configSettings.global.cards.videoPreviews', newVideos);
```
no need to spread to a new array here, `videos.filter()` already returns a new array.

File: pep/tests/integration/pods/components/modal-dialogs/admin/video/component-test.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        assert.equal(this.element.textContent?.trim(), '');
+
+        // Template block usage:
+        await render(hbs`
+      {{#modal-dialogs/admin/video}}
+        template block text
+      {{/modal-dialogs/admin/video}}
+    `);
+
+        assert.equal(this.element.textContent?.trim(), 'template block text');
```
Comment (billdami) [inline pep/tests/integration/pods/components/modal-dialogs/admin/video/component-test.ts:26]:
```suggestion
        
```
test is prob failing

File: pep/types/@gavant/ember-button-spinner/component.d.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import { BaseGlimmerSignature } from 'pep/utils/types';
+
+import type { ButtonSpinnerArgs } from '@gavant/ember-button-spinner/components/button-spinner';
+export declare class ButtonSpinnerComponent extends Component<BaseGlimmerSignature<ButtonSpinnerArgs>> {}
+
+declare module '@glint/environment-ember-loose/registry' {
+    export default interface Registry {
+        ButtonSpinner: typeof ButtonSpinnerComponent;
+    }
+}
```
Comment (billdami) [inline pep/types/@gavant/ember-button-spinner/component.d.ts:12]: we should probably port these over to the actual addons if possible, when we get a chance.

**Review by billdami — CHANGES_REQUESTED** [review]

File: pep/types/@gavant/ember-button-spinner/component.d.ts (hunk 4 above)
Comment (bakerac4) [reply pep/types/@gavant/ember-button-spinner/component.d.ts:12]: I have mixed feelings about that, but we can discuss in chat. Resolving for now though.

**Review by billdami — APPROVED** [review]

**Pull request merged**